src/mds/Locker.cc

```diff
@@ -2991,6 +2992,10 @@ void Locker::_update_cap_fields(CInode *in, int dirty, MClientCaps *m, inode_t *
 	      << " for " << *in << dendl;
       pi->mtime = mtime;
     }
+    if (((dirty & CEPH_CAP_FILE_WR) && change_attr > pi->change_attr) ||
+	((dirty & CEPH_CAP_FILE_EXCL) && change_attr != pi->change_attr)) {
+      pi->change_attr = change_attr;
```

Comment:
Yeah, that may be the safest way to handle it, but I'm not sure this is the right spot to do that. It may also legitimately not have changed here because a client that does support the changeid field had the caps and nothing actually changed.

What we probably want to do is detect when we have an older client and then just preemptively bump the changeid in that situation.  I'll have to think about that piece.